test/e2e/framework.go

```diff
@@ -972,6 +972,24 @@ func (data *TestData) createNginxPodOnNode(name string, nodeName string) error {
 	}, false, nil)
 }
 
+// createAgnhostPodOnNodeWithMultiPort creates a Pod in the test namespace with
+// a single agnhost container listening multiple ports.
+// The Pod will be scheduled on the specified Node (if nodeName is not empty).
+func (data *TestData) createAgnhostPodOnNodeWithMultiPort(name string, nodeName string, ports []int32) error {
+	var args []string
+	var containerPorts []corev1.ContainerPort
+	for _, port := range ports {
+		args = append(args, fmt.Sprintf("/agnhost serve-hostname --tcp --http=false --port=%d", port))
+		containerPorts = append(containerPorts, corev1.ContainerPort{
+			Name:          fmt.Sprintf("c%d", port),
+			ContainerPort: port,
+			Protocol:      corev1.ProtocolTCP,
+		})
+	}
+
+	return data.createPodOnNode(name, nodeName, agnhostImage, []string{"/bin/bash", "-c"}, []string{strings.Join(args, " & ")}, nil, containerPorts, false, nil)
```

Comment:
I find it a bit messy. Why didn't you choose to create a different container for each port, like we do for ANP / ACNP tests?

Reply:
Changed.